File: hw/suarez/U3/HW18AbstractFactory/src/ec/edu/espe/model/WinFactory.java
package ec.edu.espe.model;

/**
 *
 * @author bernardo
 */

public class WinFactory extends GUIFactory {
    
    @Override
    public Button createButton() {
        return new WinButton();
    }

    @Override
    public Menu createMenu() {
        return new WinMenu();
    }
    
}

File: hw/paillacho/U3/HW19AbstractFactory/AbstractFactoryPython/ec/edu/espe/composite/abstractfactory/controller/WinFactory.py
"""
Windows Factory implementation

@author PC
"""
from .GUIFactory import GUIFactory
from ec.edu.espe.composite.abstractfactory.model.Button import Button
from ec.edu.espe.composite.abstractfactory.model.Menu import Menu
from ec.edu.espe.composite.abstractfactory.view.WinButton import WinButton
from ec.edu.espe.composite.abstractfactory.model.WinMenu import WinMenu


class WinFactory(GUIFactory):
    
    def create_button(self):
        return WinButton()
    
    def create_menu(self):
        return WinMenu()
